src/storage_persister.rs

```diff
@@ -253,19 +269,62 @@ impl<ChannelSigner: Sign> Persist<ChannelSigner> for StoragePersister {
         data: &ChannelMonitor<ChannelSigner>,
         update_id: MonitorUpdateId,
     ) -> ChannelMonitorUpdateStatus {
-        /*let key = channel_id.to_channel_id().to_hex();
-        let data = data.encode();
-        if self
-            .storage
-            .put_object(MONITORS_BUCKET.to_string(), key, data)
-            .is_err()
+        // Persist locally
+        match self
+            .fs_persister
+            .persist_new_channel(channel_id, data, update_id)
         {
-            return ChannelMonitorUpdateStatus::PermanentFailure;
-        }
-        ChannelMonitorUpdateStatus::Completed*/
+            ChannelMonitorUpdateStatus::Completed => {}
+            ChannelMonitorUpdateStatus::InProgress => {
+                error!("Unexpected: FilesystemPersister returned ChannelMonitorUpdateStatus::InProgress");
+                return ChannelMonitorUpdateStatus::PermanentFailure;
+            }
+            ChannelMonitorUpdateStatus::PermanentFailure => {
+                error!("Failed to persist a ChannelMonitor in the filesystem.");
+                return ChannelMonitorUpdateStatus::PermanentFailure;
+            }
+        };
 
-        self.fs_persister
-            .persist_new_channel(channel_id, data, update_id)
+        // Launch background task that handles persisting monitor remotely
+        let data = data.encode();
+        let storage = Arc::clone(&self.storage);
+        let key = channel_id.to_channel_id().to_hex();
+        let chain_monitor = match self.chain_monitor.lock().unwrap().upgrade() {
+            None => return ChannelMonitorUpdateStatus::PermanentFailure,
+            Some(c) => c,
+        };
+        self.runtime_handle.spawn(async move {
+            // The channel will block until remote persistence succeeds so let's continuously retry
+            // 
+            // If there is a non-runtime failure or if the node is shutdown before remote persistence 
+            // succeeds, the local version of the channel monitor will be more fresh than the remote one
+            // 
+            // This should sort itself out as the node will get back at trying to persist remotely
+            // when it gets back online (adding the ChannelMonitor to the ChainMonitor calls 
+            // persist_new_channel() again).
+            loop {
```

Comment:
As discussed out-of-band, let's stop retrying to persist when [list_pending_monitor_updates()](https://docs.rs/lightning/latest/lightning/chain/chainmonitor/struct.ChainMonitor.html#method.list_pending_monitor_updates) stops including the `update_id` in question. This will be reflected in #169